### ISSUE TITLE
"Actual" Minimum Stock for warehouse

### ISSUE THREAD

**Issue by XykoZ** [ldtteam/minecolonies-features#234]
### Prerequisites 

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [X] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my issue is not covered there.
- [X] I made sure that **this change request is not a duplicate of any existing change request.**

### Context 

- Minecraft Version: 1.15
- MineColonies Version: 0.13.548-ALPHA
- Related Mods and their Versions: Craft The Exile: Dissonance

## Is your feature request related to a problem?

The warehouse runs out of stock of things that the player wants to make use of sporadically.
The current Minimum Stock function allows other buildings to consume them, which means that the warehouse doesn't actually have a "minimum stock" of them.
If I set the warehouse to have a minimum stock of 1 stack of something, I expect it to actually keep 1 stack on hand, not give it away.

## Enhancement or Change description

I appreciate the current minimum stock function, and it's fine, it can stay (or be renamed to something else depending on definition). This is an enhancement where you can actually set a minimum stock that the warehouse *actually* keeps in stock and doesn't give away. This is required when preparing for upcoming research, upgrades, buildings, etc. You can set a minimum stock that you can rely on.

### Notes or related things

---
## Viewers

* Add a thumbs-up to the change request to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by PowerSchill** [comment]
The workaround for this would be to drop a postbox right outside of your university. Then you can request items you need for research and they won't be used for other things.

**Comment by jzisser9** [comment]
> The workaround for this would be to drop a postbox right outside of your university. Then you can request items you need for research and they won't be used for other things. 

This should work. The only downside is, I believe, the request is cleared from the queue once a courier delivers it, so you would need to request another stack of the material through the postbox afterwards.

**Comment by wrincewind** [comment]
if it's for a university, then you should only need it once - when a research is done, it's done. If it's for things you need all the time (e.g. wool for banners on fortress-type colonies), then yeah, this doesn't work so well. 

An alternate suggestion would be a 'keep-in-stock' mode for the postbox that re-fills it if it ever dips below the required levels, like the warehouse's current minimum-stock mode.
